### PR TITLE
fix - rate limiter & duplicated methods & env variables

### LINKED ISSUE #36: rate limiter should be configured to ignore WebSocket and SSE
Hi everyone,

I noticed on one of my latest projects that the rate limiter is affecting WebSocket and SSE connections. In order to improve the user experience and ensure smooth real-time communication, I suggest configuring the rate limiter to ignore WebSocket and SSE requests. This will allow uninterrupted data streaming and reduce unnecessary restrictions.

### LINKED ISSUE #70: fix - Removing duplicated methods of role and authorities validation

### PR DESCRIPTION
**Pull request by bellaabdelouahab**
Task done :

- [X] updates env variable to be accessed from env.example  in case the env not exist, also remove some hard coded credential 
- [X] add two built-in methods for user model to verify existence of an action in authorities or restrictions 
- [X] configure rate limiting to ignore SSE connections 
- [X] disable generating the docs each time the server starts ( just for the moment till we create a separate npm command for it  )
- [X] removing duplicated methods for role validation : 
     -  restrictedTo in https://github.com/ISIL-ESTE/Student-Workflow-Organizer/blob/0140aa1fdfd243d112af7ab2addb899b99112f4c/backend-app/controllers/auth_controller.js#L324  is used to validated roles 
     - and the two built in methods are used to validate actions ( use it wisely 😁 )

 
 related issues :

-  resolves #70 
- resolves #36

### CODE REVIEW

File: backend-app/models/user_model.js [hunk 1]
```diff
@@ -92,6 +92,16 @@ userSchema.methods.correctPassword = async function (
...
 };
 
+
+// verify if the user is authorized or restricted from an action
+userSchema.methods.isAuthorizedTo = function (action) {
+  return this.authorities.includes(action);
+};
+userSchema.methods.isRestrictedFrom = function (action) {
+  return this.restrictions.includes(action);
+};
```
Comment (muttaqin1) [inline backend-app/models/user_model.js:102]: 👌👌👌

**Review by muttaqin1 — COMMENTED** [review]
Great job 💯

File: backend-app/.env.example [hunk 2]
```diff
@@ -1,8 +1,13 @@
+NODE_ENV = "development"
 API_VERSION = "v1"
-MONGO_URI = "mongodb+srv://username:password@cluster0.cmlpjag.mongodb.net/?retryWrites=true&w=majority"
+MONGO_URI = "mongodb+srv://swo:zrcoCze9UKFTWzYx@nfs315.tayahgd.mongodb.net/S-W-O"
+PORT = 5000
 ADMIN_EMAIL = "admin@swf.com"
 ADMIN_PASSWORD = "password123418746"
 JWT_SECRET = "JWT-Seceret-Key"
```
Comment (Khalid1G) [inline backend-app/.env.example:7]: Use this to generate a real JWT_SECRET : 
```shel
node -e "console.log(require('crypto').randomBytes(128).toString('hex'))"
```

**Review by Khalid1G — APPROVED** [review]
well

**Review by Khalid1G — APPROVED** [review]

File: backend-app/.env.example (hunk 2 above)
Comment (bellaabdelouahab) [reply backend-app/.env.example:7]: sure

**Pull request merged**